packages/core/src/react-integration/provider/BackupBoundary.tsx

```diff
@@ -1,15 +1,26 @@
 import React, { Suspense, memo, useMemo, version } from 'react';
 
 /* istanbul ignore next  */
-const NoSuspense =
-  (version.startsWith('16') || version.startsWith('17')) &&
-  typeof window === 'undefined';
+const LegacyReact = version.startsWith('16') || version.startsWith('17');
```

Comment:
the NoSuspense check, and this one, are super brittle - it'd be much better to feature-detect rather than useragent-sniff, if at all possible.

Reply:
I agree feature detection is much better; but I can't find any indicators on google, and using the actual feature itself (by trying to render suspense) would break everything so that wouldn't work either. I assume when you say 'useragent sniff' you are drawing and analogy to browser behavior and are not actually thinking that the 'version' export of react is a browser user-agent.

Reply:
lol yes, exactly what i meant :-)

def not a blocker if there's no better way for now, ofc

Reply:
Given that these behaviors with major versions are pretty well defined; and multiple react versions are not possible because they break so many other things - I don't see this actually being much of an issue.